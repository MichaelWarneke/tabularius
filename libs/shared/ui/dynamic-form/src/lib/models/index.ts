import {
  FormControl,
  ValidatorFn,
  AbstractControlOptions,
  AsyncValidatorFn,
  FormArray
} from '@angular/forms';

export class FormControlBase extends FormControl {
  controlType = '';
  key = '';
  label = '';
  maxWidth = 100;
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlTextbox extends FormControlBase {
  controlType = 'input';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlDate extends FormControlBase {
  controlType = 'datepicker';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlCheckbox extends FormControlBase {
  controlType = 'checkbox';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlSelect extends FormControlBase {
  controlType = 'select';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlRadioButton extends FormControlBase {
  controlType = 'radio-button';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export class FormControlArray extends FormArray {
  controlType = 'array';
  key = '';
  label = '';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export { Validators } from '@angular/forms';
