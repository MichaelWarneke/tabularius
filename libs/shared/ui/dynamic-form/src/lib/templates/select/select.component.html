<mat-form-field class="tabu-select" [formGroup]="group">
  <mat-select [placeholder]="control?.label" [formControlName]="control?.key">
    <mat-option
      class="tabu-option-item"
      *ngFor="let item of control?.options"
      [value]="item"
      >{{ item }}</mat-option
    >
  </mat-select>
</mat-form-field>
