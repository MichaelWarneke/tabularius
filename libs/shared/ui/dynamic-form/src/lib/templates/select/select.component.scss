.tabu-select {
  margin: 10px;
}
