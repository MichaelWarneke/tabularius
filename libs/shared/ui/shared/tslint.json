{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
