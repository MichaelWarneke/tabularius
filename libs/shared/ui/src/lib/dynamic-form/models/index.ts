export { QuestionBase } from './question-base';
export { DropdownQuestion } from './question-dropdown';
export { TextboxQuestion } from './question-textbox';
export { DatepickerQuestion } from './question-datepicker';
