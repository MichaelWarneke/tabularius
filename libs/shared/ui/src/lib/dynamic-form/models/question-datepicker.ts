import { QuestionBase } from './question-base';

export class DatepickerQuestion extends QuestionBase<string> {
  controlType = 'datepicker';
  type: Date;

  constructor(options: any = {}) {
    super(options);
    this.type = options['type'] || '';
  }
}
