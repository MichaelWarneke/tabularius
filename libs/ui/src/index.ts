export { TabuAppSidenavModule } from './lib/nav';
