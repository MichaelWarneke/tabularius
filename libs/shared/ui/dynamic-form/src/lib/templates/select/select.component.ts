import { Component } from '@angular/core';
import { FormControlSelect } from '../../models';
import { BaseComponent } from '../base.component';

@Component({
  selector: 'tabu-dynamic-form-select',
  templateUrl: './select.component.html',
  styleUrls: ['./select.component.scss']
})
export class SelectComponent extends BaseComponent {
  control: FormControlSelect | null = null;
  constructor() {
    super();
  }
}
