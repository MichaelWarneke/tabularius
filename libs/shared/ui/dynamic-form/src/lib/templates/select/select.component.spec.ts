import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { SelectComponent } from './select.component';
import { FormGroup, FormControl } from '@angular/forms';
import { FormControlSelect } from '../../models';
import { SharedModule } from '@tabularius/shared/ui';
import { By } from '@angular/platform-browser';

describe('SelectComponent', () => {
  let component: SelectComponent;
  let fixture: ComponentFixture<SelectComponent>;
  const key = 'testKey';
  const label = 'testLabel';
  const options = ['one', 'two', 'three'];

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule, NoopAnimationsModule],
      declarations: [SelectComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SelectComponent);
    component = fixture.componentInstance;
    component.group = new FormGroup({ testKey: new FormControl('') });
    component.control = new FormControlSelect();
    component.control.key = key;
    component.control.label = label;
    component.control.options = options;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain the label', () => {
    const attributes = fixture.debugElement.query(By.css('mat-select'))
      .attributes;
    const text = attributes['placeholder'];
  });

  it('should contain the formControlName in attributes', () => {
    const attributes = fixture.debugElement.query(By.css('mat-select'))
      .attributes;
    const text = attributes['ng-reflect-name'];
    expect(text).toContain(key);
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
