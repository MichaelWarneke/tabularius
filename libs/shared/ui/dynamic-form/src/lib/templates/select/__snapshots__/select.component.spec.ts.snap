// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectComponent should render 1`] = `
<tabu-dynamic-form-select
  control={[Function FormControlSelect]}
  group={[Function FormGroup]}
>
  <mat-form-field
    class="tabu-select mat-form-field ng-tns-c12-0 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
    ng-reflect-form="[object Object]"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <mat-select
            aria-disabled="false"
            aria-invalid="false"
            aria-label="testLabel"
            aria-multiselectable="false"
            aria-required="false"
            class="mat-select ng-tns-c13-1 mat-select-empty ng-untouched ng-pristine ng-valid ng-star-inserted"
            id="mat-select-3"
            ng-reflect-name="testKey"
            ng-reflect-placeholder="testLabel"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-hidden="true"
              cdk-overlay-origin=""
              class="mat-select-trigger"
            >
              <div
                class="mat-select-value"
                ng-reflect-ng-switch="true"
              >
                
                <span
                  class="mat-select-placeholder ng-tns-c13-1 ng-star-inserted"
                >
                  testLabel
                </span>
                
              </div>
              <div
                class="mat-select-arrow-wrapper"
              >
                <div
                  class="mat-select-arrow"
                />
              </div>
            </div>
            
          </mat-select>
          <span
            class="mat-form-field-label-wrapper"
          >
            
            <label
              aria-owns="mat-select-3"
              class="mat-form-field-label ng-tns-c12-0 mat-empty mat-form-field-empty ng-star-inserted"
              for="mat-select-3"
              id="mat-form-field-label-7"
              ng-reflect-disabled="true"
              ng-reflect-ng-switch="false"
            >
              
              
              testLabel
              
              
            </label>
          </span>
        </div>
        
      </div>
      
      <div
        class="mat-form-field-underline ng-tns-c12-0 ng-star-inserted"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
        ng-reflect-ng-switch="hint"
      >
        
        
        <div
          class="mat-form-field-hint-wrapper ng-tns-c12-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field>
</tabu-dynamic-form-select>
`;
